### PR TITLE
redirect validator dashboard with query params

### PR DESCRIPTION
**Pull request by MauserBitfly**
This PR:
- implements redirecting of the validator dashboard if the validator list comes as query params

Example:
/dashboard?validators=5,0x832b8286f5d6535fd941c6c4ed8b9b20d214fc6aa726ce4fba1c9dbb4f278132646304f550e557231b6932aa02cf08d3
-> /dashboard/NSwweDgzMmI4Mjg2ZjVkNjUzNWZkOTQxYzZjNGVkOGI5YjIwZDIxNGZjNmFhNzI2Y2U0ZmJhMWM5ZGJiNGYyNzgxMzI2NDYzMDRmNTUwZTU1NzIzMWI2OTMyYWEwMmNmMDhkMw#summary

### CODE REVIEW

File: frontend/middleware/redirect.global.ts [hunk 1]
```diff
@@ -36,6 +36,18 @@ export default function ({ name, params }: RouteLocationNormalizedLoaded) {
         `${v1Domain}/address/${params.id || params.slug?.[1]}`,
         { external: true }
       )
+    case 'dashboard':
+    case 'dashboard-id':
+      if (query.validators && typeof query.validators === 'string') {
+        const list = query.validators.split(',').filter((v) => {
+          return !isNaN(parseInt(v)) || isPublicKey(v)
```
Comment (thib-wien) [inline frontend/middleware/redirect.global.ts:43]: `!isNaN(parseInt(v))` is permissive. A chain like _1234abc_ passes the test.
`!isNaN(Number(v)) && String(Number(v)) === v` would be uncheatable I think.

File: frontend/utils/regexp.ts [hunk 2]
```diff
@@ -3,3 +3,5 @@ export const REGEXP_HAS_NUMBERS = /^(?!0+$)\d+$/
 export const REGEXP_VALID_EMAIL = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/i
 
 export const REGEXP_VALID_NAME = /^(?=.*[a-zA-Z0-9])[a-zA-Z0-9 ]*$/
+
+export const REGEXP_PUBLIC_KEX = /^(1|3)[A-Za-z0-9]{39}$/
```
Comment (thib-wien) [inline frontend/utils/regexp.ts:7]: I am not an expert of regular expressions.
I believe that this one is not what we want and I asked ChatGPT 4o as a second opinion.
I think that we want
`/^(|0x)[A-Fa-f0-9]{96}$/`

**Review by thib-wien — COMMENTED** [review]
gut gemacht of course

File: frontend/utils/regexp.ts (hunk 2 above)
Comment (MauserBitfly) [reply frontend/utils/regexp.ts]: your AI is better then my AI ;-)

File: frontend/middleware/redirect.global.ts (hunk 1 above)
Comment (MauserBitfly) [reply frontend/middleware/redirect.global.ts]: fixed and added a isInt() utils function

**Review by thib-wien — APPROVED** [review]
:)

**Pull request merged**